tasks/install.js

```diff
@@ -0,0 +1,81 @@
+module.exports = function(grunt) {
+
+  /**
+   * Define tasks to quickly get Drupal running.
+   *
+   * grunt serve
+   *   Run Drupal via the PHP webserver and activate watch tasks.
+   * grunt serve:demo
+   *   Run Drupal via the PHP webserver without running watch tasks.
+   * grunt serve:test
+   *   Run Drupal via the PHP webserver without running watch tasks or opening
+   *   the browser.
+   */
+  grunt.loadNpmTasks('grunt-drush');
+
+  var Help = require('../lib/help')(grunt),
+    Drupal = require('../lib/drupal')(grunt);
+
+  var path = require('path'),
+    _ = require('lodash');
+
+  // If no path is configured for Drush, fallback to the system path.
+  var cmd = {cmd: Drupal.drushPath()};
+
+  var profile = grunt.config('config.project.profile') || 'standard';
+
+  var dbUrl = grunt.option('db-url') || false;
+  if (dbUrl) {
+    dbUrl = '--db-url=' + dbUrl;
+  }
+
+  grunt.config(['drush', 'install'], {
+    args: ['site-install', '-yv', profile, dbUrl],
+    options: _.extend({
+      cwd: '<%= config.buildPaths.html %>'
+    }, cmd)
+  });
+  grunt.config(['drush', 'sql-drop'], {
+    args:['sql-drop', '-y'],
+    options: _.extend({
+      cwd: '<%= config.buildPaths.html %>'
+    }, cmd)
+  });
+  grunt.config(['shell', 'loaddb'], {
+    command: 'gzip -dc <%= config.project.db %> | <%= config.project.dbConnection %>'
+  });
+
+  grunt.registerTask('install', 'Install Drupal with the configured profile, or from a database dump file if one exists.', function() {
+
+    var done = this.async();
+
+    // Check for a database file first.
+    var dbPath = grunt.config('config.install.db');
+    if (dbPath && grunt.file.exists(dbPath)) {
+      Drupal.loadDatabaseConnection(function(error) {
+        if (error) {
+          grunt.fail.fatal(error);
+          done();
+        }
+        grunt.task.run([
+          // Drop current database.
+          'drush:sql-drop',
+          // Load new database.
+          'shell:loaddb'
```

Comment:
For a complete db load, I generally find the following next steps should be followed:

1. If using Features module: disable features rebuild on cache clear
2. Clear cache
3. Run updatedb to ensure database is synchronized with code.
4. If features module is used, run features-revert-all to ensure database is synchronized with code.

Could we somehow build that in here, or if all those steps seem unreasonable, could we trigger a grunt event so projects can tailor the process?

This seems fine as a follow-up.